### PR TITLE
FlashDevelop startup optimization

### PR DESCRIPTION
**Pull request by elsassph**
- Delayed FlashConnect connection
- Delayed DockContent creation for rarely used UIs - applied to Profiler, FlashDebugger, FlashLogs Viewer.
- Dock panels now default to hidden.
- Show Project panel on first run (better first run experience)

### CODE REVIEW

File: FlashDevelop/Docking/DockablePanel.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public DockablePanel(Control ctrl, String pluginGuid)
             this.HideOnClose = true;
             this.Controls.Add(ctrl);
             Globals.MainForm.ThemeControls(this);
-            this.Show();
+            //this.Show();
```
Comment (elsassph) [inline FlashDevelop/Docking/DockablePanel.cs]: Seem to work fine with NOT showing panels by default and letting the layout handle it.

**Comment by Meychi** [comment]
Should this be still worked on?

**Comment by elsassph** [comment]
I proposed to do this optimization to try reducing startup time. I also wanted to hide rarely needed panels so that the first run experience looks cleaner.

At the very least, FlashConnect startup should be delayed significantly.

**Comment by Meychi** [comment]
I think also "Show ProjectPanel on firstRun" is important. I think the debugger panels should be docked together. Otherwise i think the initial ui is ok. I suggest we do:

"Show ProjectPanel on firstRun"  & FlashConnect startup delayed & debugger panels docked together.

**Comment by elsassph** [comment]
Ok, are you up to extracting it or should I do it?

**Comment by Meychi** [comment]
I can do this.

**Comment by Meychi** [comment]
Done.

**Pull request closed without merging**